test/dotnet-publish.Tests/Microsoft.DotNet.Tools.Publish.Tests.cs

```diff
@@ -26,35 +54,30 @@ public static IEnumerable<object[]> PublishOptions
             {
                 return new[]
                 {
-                    new object[] { "", "", "", "" },
-                    new object[] { "dnxcore50", "", "", "" },
-                    new object[] { "", PlatformServices.Default.Runtime.GetLegacyRestoreRuntimeIdentifier(), "", "" },
-                    new object[] { "", "", "Release", "" },
-                    new object[] { "", "", "", "some/dir"},
-                    new object[] { "", "", "", "some/dir/with spaces" }, 
-                    new object[] { "dnxcore50", PlatformServices.Default.Runtime.GetLegacyRestoreRuntimeIdentifier(), "Debug", "some/dir" },
+                    new object[] { "1", "", "", "", "" },
+                    new object[] { "2", "dnxcore50", "", "", "" },
+                    new object[] { "3", "", PlatformServices.Default.Runtime.GetLegacyRestoreRuntimeIdentifier(), "", "" },
+                    new object[] { "4", "", "", "Release", "" },
+                    new object[] { "5", "", "", "", "some/dir"},
+                    new object[] { "6", "", "", "", "some/dir/with spaces" },
+                    new object[] { "7", "dnxcore50", PlatformServices.Default.Runtime.GetLegacyRestoreRuntimeIdentifier(), "Debug", "some/dir" },
                 };
             }
         }
 
         [Theory]
         [MemberData("PublishOptions")]
-        public void PublishOptionsTest(string framework, string runtime, string config, string outputDir)
+        public void PublishOptionsTest(string testIdentifier, string framework, string runtime, string config, string outputDir)
         {
-            // create unique directories in the 'temp' folder
-            var root = Temp.CreateDirectory();
+            TestScenario scenario = TestScenario.Create(Path.Combine(_testProjectsRoot, "TestAppWithLibrary"), skipRestore: true);
+            TestInstance instance = scenario.CreateTestInstance(identifier: testIdentifier)
```

Comment:
TestScenario.Create("TestAppWithLibrary);
Isn't that sufficient? Restore and build should already be done and the repo root is common to all tests...